### PR TITLE
DGNSS command line runner.

### PR DESCRIPTION
**Pull request by mookerji**
A standalone command line tool for running the RTK filters. The
intention here is that a separate tool produces post-processed
observations which are then consumed by this one.

Mostly a WIP. I'd like to reincorporate the reporting stuff.

/cc @imh @fnoble

### CODE REVIEW

File: gnss_analysis/dgnss.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+"""
+
+from gnss_analysis.constants import MIN_SATS
+import pandas as pd
+import swiftnav.dgnss_management as mgmt
+
+
+def process_dgnss(table):
+  """Software simulate the RTK filter given a GPS timestamped
```
Comment (fnoble) [inline gnss_analysis/dgnss.py]: IMHO "simulate" is misleading for what SITL is doing? I feel like it implies we are using fake data or we are not running the same code.

File: gnss_analysis/dgnss.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  observations from a HITL test. See:
```
Comment (fnoble) [inline gnss_analysis/dgnss.py]: Maybe remove "from a HITL test"? The observations could come from elsewhere?

File: gnss_analysis/dgnss.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    if not (rover_spp_sim_t is None or base_spp_sim_t is None):
+      n_sds = len(rover_sdiff_t)
+      if not init_done and n_sds >= MIN_SATS:
+        print "Initializing DGNSS!"
+        mgmt.dgnss_init(rover_sdiff_t, rover_spp_sim_t.pos_ecef)
+        init_done = True
+      elif n_sds >= MIN_SATS:
+        mgmt.dgnss_update(rover_sdiff_t, rover_spp_sim_t.pos_ecef)
+        num_used, b = mgmt.dgnss_fixed_baseline(rover_sdiff_t,
+                                                base_spp_sim_t.pos_ecef)
```
Comment (fnoble) [inline gnss_analysis/dgnss.py]: Don't currently do anything with `b` and `num_used`? Also how do we expose filter state?

**Comment by fnoble** [comment]
Looks good, lets sync up with @imh and decide on terminology. Also, would be interested to discuss how this fits with the analysis stuff.

File: gnss_analysis/dgnss.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+def get_filter_state():
+  initial_sats = mgmt.get_sats_management()[1]
+  kf_means = mgmt.get_amb_kf_mean()
+  kf_covs = mgmt.get_amb_kf_cov2()
+  iar_ambs = mgmt.dgnss_iar_MLE_ambs()
+  print iar_ambs
+
+def process_dgnss(table):
+  """Software simulate the RTK filter given a GPS timestamped
```
Comment (imh) [inline gnss_analysis/dgnss.py]: As fergus said before, simulate is the wrong idea. It's more like postprocessing or reprocessing saved data.

File: gnss_analysis/dgnss.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  ----------
+  table : pandas.io.pytables.HDFStore
+    Pandas table
+
+  """
+  assert isinstance(table, pd.HDFStore), "Not a Pandas table!"
+  init_done = False
+  results = {}
+  for timestamp, sdiff_t in table.sdiffs.iteritems():
+    rover_spp_sim_t = table.rover_spp_sim.get(timestamp, None)
```
Comment (imh) [inline gnss_analysis/dgnss.py]: 'sim' is simulated?

File: gnss_analysis/tools/simulate_dgnss.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+#          Bhaskar Mookerji <mookerji@swiftnav.com>
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+"""This tool runs a software-in-the-loop (SITL) simulation of the
```
Comment (imh) [inline gnss_analysis/tools/simulate_dgnss.py]: same sim stuff. It's not a simulation, and also (hopefully) useful for postprocessing.

File: gnss_analysis/dgnss.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        mgmt.dgnss_init(sdiff_t, base_spp_sim_t.values)
+        init_done = True
+      elif n_sds >= MIN_SATS:
+        get_filter_state()
+        mgmt.dgnss_update(sdiff_t, base_spp_sim_t.values)
+        b = None
+        num_used, b = mgmt.dgnss_fixed_baseline(sdiff_t, base_spp_sim_t.values)
+        #print b
+        results[timestamp] = b
+  return results
```
Comment (imh) [inline gnss_analysis/dgnss.py]: I don't see why baselines are particularly special as an output. I rarely compute them in SITL runs myself.

File: gnss_analysis/tools/simulate_dgnss.py [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  parser.add_argument('file',
+                      help='Specify the log file to use.')
+  args = parser.parse_args()
+  log_datafile = args.file
+  with pd.HDFStore(log_datafile) as store:
+    # assert validate_table_schema(store, _EXPECTED_KEYS), \
+    #     "Invalid schema! %s" % store.keys()
+    try:
+      baselines = process_dgnss(store)
+      store.put('rover_rtk_ecef_sim', pd.DataFrame(baselines))
```
Comment (imh) [inline gnss_analysis/tools/simulate_dgnss.py]: Same as above. If we have something special to processing baselines, we should specify that in the file name or function name or something.

**Comment by imh** [comment]
We have a bunch of general purpose stuff for getting things out of the filter. Why all this new stuff for baselines, instead of just passing the baseline reports to the old runner?

**Pull request merged**